src/components/SelectedPokemon/SelectedPokemon.tsx

```diff
@@ -30,7 +28,7 @@ class SelectedPokemonPure extends React.Component<Props> {
   }
 
   checkFavorite = (id) => {
-    const favorites = this.props.pokemon.favorites;
+    const favorites = Array.from(this.props.favoritesPokemons);
     let idArray = [];
     favorites.forEach(elem => {
       idArray.push(elem.id);
```

Comment:
tutaj zamiast robić forEacha i pusha, można to zrobić funkcyjnie

```
 const idArray = favorites.map(favorite => favorite.id);
```

efekt jest taki sam, a kod jest bardziej czytelny i generalnie porpagujemy programowanie raczej deklaratywne

tutaj wiecej na ten temat:
https://medium.com/javascript-scene/master-the-javascript-interview-what-is-functional-programming-7f218c68b3a0